packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx

```diff
@@ -333,9 +428,73 @@ function MapAnalysisImpl(props: ImplProps) {
     selectedOverlayVariable: activeMarkerConfiguration?.selectedVariable,
     overlayConfig: activeOverlayConfig.value,
     outputEntityId: outputEntity?.id,
-    //TO DO: maybe dependentAxisLogScale
+    dependentAxisLogScale:
+      activeMarkerConfiguration &&
+      'dependentAxisLogScale' in activeMarkerConfiguration
+        ? activeMarkerConfiguration.dependentAxisLogScale
+        : false,
   });
 
+  const { markersData: previewMarkerData } = useStandaloneMapMarkers({
+    boundsZoomLevel: undefined,
+    geoConfig: geoConfig,
+    studyId,
+    filters,
+    markerType,
+    selectedOverlayVariable: activeMarkerConfiguration?.selectedVariable,
+    overlayConfig: activeOverlayConfig.value,
+    outputEntityId: outputEntity?.id,
+  });
+
+  const continuousMarkerPreview = useMemo(() => {
+    if (!previewMarkerData || !previewMarkerData.length) return;
+    const initialDataObject = previewMarkerData[0].data.map((data) => ({
+      label: data.label,
+      value: 0,
+      ...(data.color ? { color: data.color } : {}),
+    }));
+    const typedData =
+      markerType === 'pie'
+        ? ([...previewMarkerData] as DonutMarkerProps[])
+        : ([...previewMarkerData] as ChartMarkerProps[]);
+    const finalData = typedData.reduce(
+      (prevData, currData) =>
+        currData.data.map((data, index) => ({
+          label: data.label,
+          value: data.value + prevData[index].value,
+          ...('color' in prevData[index]
+            ? { color: prevData[index].color }
+            : 'color' in data
+            ? { color: data.color }
+            : {}),
+        })),
+      initialDataObject
+    );
+    if (markerType === 'pie') {
+      return (
+        <DonutMarkerStandalone
+          data={finalData}
+          markerLabel={kFormatter(finalData.reduce((p, c) => p + c.value, 0))}
+          {...sharedStandaloneMarkerProperties}
+        />
+      );
+    } else {
+      return (
+        <ChartMarkerStandalone
+          data={finalData}
+          markerLabel={mFormatter(finalData.reduce((p, c) => p + c.value, 0))}
+          dependentAxisLogScale={
+            activeMarkerConfiguration &&
+            'dependentAxisLogScale' in activeMarkerConfiguration
+              ? activeMarkerConfiguration.dependentAxisLogScale
+              : false
+          }
+          {...sharedStandaloneMarkerProperties}
+        />
+      );
+    }
+  }, [previewMarkerData]);
```

Comment:
It may be possible to move the logic for generating `continuousMarkerPreview` inside the `*MarkerConfigurationMenu` components (as a utility function, I imagine). It looks like all but one (`geoConfig`) of the props needed to make `previewMarkerData` are already passed to the `*MarkerConfigurationMenu` components, so it wouldn't involve a lot of "prop bloat". In fact we'd be removing one prop (`continuousMarkerPreview`) and adding `geoConfig`, so breaking even!

It would clean things up a bit at this outer level.

Reply:
I wonder if this should wait for https://github.com/VEuPathDB/web-monorepo/issues/106?